### DISCUSSION TITLE
Wavetable browsing improvement

### DISCUSSION THREAD

**Discussion by heeen** [SynthstromAudible/DelugeFirmware#243] [Ideas]
Currently browsing wavetables is sort of tedious. You have no idea how the oscillator will sound as the whole sample is just played through once. The waveform rendering is also kind of useless: wavetables tend to have the same amplitude along the timeline when squished to the screen space available.

Suggestion:

Play the wavetable like a wavetable, that is set up a wavetable oscillator and have it ping pong through the file, while also displaying only the current cycle stretched to the screen width, letting the user see the waveform morph through its range. Optionally, commandeer one of the encoders to stop the ping pong and move the window manually instead.

**Comment by manysounds** [comment]
0-50 as a wavetable index position is insufficient, among some other potential wavetable issues.
A correctly generated wavetable has the block size in the data so dividing the size of the table by the block size should give us a number to scroll through instead of 0-50. This would give us 0-x grains to choose from.
Some wavetables (especially older ones) and user generated unknown files have no block size. 
The default size of 2048 may be incorrect in that case.
This would require an expanded import wavetable step that looks for repeats and automagically picks the block size (completely unlike the multi-sample import but just as clever) or a user driven wavetable import method that allows the user to specify the block size during the import into instrument step.
Ideally, we only have a 0-x for the actual number of wave grains in a table even if the file has no information.
Orrrrr I don’t know what I’m talking about, which happens.

**Comment by legsmechanical** [comment]
Big plus one on the wavetable preview. As it is, I just don’t use the wavetables bc their so cumbersome to audition (which is a huge shame).
